File: src/main/java/tn/esprit/models/AnnonceMatch.java
package tn.esprit.models;


import java.time.LocalDateTime;


public class AnnonceMatch {
    private int annonceId;
    private String titre;
    private LocalDateTime date_Heure;
    private String lieu;
    private int joueursRecherches;
    private String niveau;
    private String description;

    public AnnonceMatch(int annonceId, String titre, LocalDateTime date_Heure, String lieu, int joueursRecherches, String niveau, String description) {
        this.annonceId = annonceId;
        this.titre = titre;
        this.date_Heure = date_Heure;
        this.lieu = lieu;
        this.joueursRecherches = joueursRecherches;
        this.niveau = niveau;
        this.description = description;
    }
    public AnnonceMatch( String titre, LocalDateTime date_Heure, String lieu, int joueursRecherches, String niveau, String description) {
        this.titre = titre;
        this.date_Heure = date_Heure;
        this.lieu = lieu;
        this.joueursRecherches = joueursRecherches;
        this.niveau = niveau;
        this.description = description;
    }

    public int getAnnonceId() {
        return annonceId;
    }

    public void setAnnonceId(int annonceId) {
        this.annonceId = annonceId;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public LocalDateTime getDate_Heure() {
        return date_Heure;
    }

    public void setDate_Heure(LocalDateTime dateHeure) {
        this.date_Heure = date_Heure;
    }

    public String getLieu() {
        return lieu;
    }

    public void setLieu(String lieu) {
        this.lieu = lieu;
    }

    public int getJoueursRecherches() {
        return joueursRecherches;
    }

    public void setJoueursRecherches(int joueursRecherches) {
        this.joueursRecherches = joueursRecherches;
    }

    public String getNiveau() {
        return niveau;
    }

    public void setNiveau(String niveau) {
        this.niveau = niveau;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "AnnonceMatch{" +
                "annonceId=" + annonceId +
                ", titre='" + titre + '\'' +
                ", dateHeure=" + date_Heure +
                ", lieu='" + lieu + '\'' +
                ", joueursRecherches=" + joueursRecherches +
                ", niveau='" + niveau + '\'' +
                ", description='" + description + '\'' +
                '}';
    }




}

File: src/Entity/Annoncematch.php
<?php

namespace App\Entity;

use App\Repository\AnnonceMatchRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: AnnonceMatchRepository::class)]
#[ORM\Table(name: 'annoncematch')]
class AnnonceMatch
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'annonce_id')]
    private ?int $annonceId = null;

    #[ORM\Column(length: 25)]
    private ?string $titre = null;

    #[ORM\Column(name: 'date_heure', type: Types::DATE_MUTABLE)]
    private ?\DateTimeInterface $dateHeure = null;

    #[ORM\Column(length: 25, nullable: true)]
    private ?string $lieu = null;

    #[ORM\Column(name: 'joueurs_recherches')]
    private ?int $joueursRecherches = null;

    #[ORM\Column(length: 25)]
    private ?string $niveau = null;

    #[ORM\Column(length: 25)]
    private ?string $description = null;

    public function getAnnonceId(): ?int
    {
        return $this->annonceId;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): static
    {
        $this->titre = $titre;

        return $this;
    }

    public function getDateHeure(): ?\DateTimeInterface
    {
        return $this->dateHeure;
    }

    public function setDateHeure(\DateTimeInterface $dateHeure): static
    {
        $this->dateHeure = $dateHeure;

        return $this;
    }

    public function getLieu(): ?string
    {
        return $this->lieu;
    }

    public function setLieu(?string $lieu): static
    {
        $this->lieu = $lieu;

        return $this;
    }

    public function getJoueursRecherches(): ?int
    {
        return $this->joueursRecherches;
    }

    public function setJoueursRecherches(int $joueursRecherches): static
    {
        $this->joueursRecherches = $joueursRecherches;

        return $this;
    }

    public function getNiveau(): ?string
    {
        return $this->niveau;
    }

    public function setNiveau(string $niveau): static
    {
        $this->niveau = $niveau;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;

        return $this;
    }
} 